source/pem_utils.c

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/common/string.h>
+#include <aws/io/private/pem_utils.h>
+
+enum aws_pem_util_state {
+    BEGIN,
+    ON_DATA,
+    END,
+};
+
+int aws_clean_up_pem(struct aws_byte_buf *pem, struct aws_allocator *allocator) {
```

Comment:
I think we might want to name this `aws_normalize_pem` or `aws_sanitize_pem`, clean_up is the wrong verb, since clean_up means something else throughout the rest of our code.